### PR TITLE
Properly transform between camera and telescope frame in muon fitter

### PR DESCRIPTION
**Pull request by maxnoe**
The muon code did some strange things to transform pixel coordinates from camera to telescope frame, probably from code before camera geometries could be transformed as a whole.

### CODE REVIEW

File: ctapipe/image/muon/tests/test_intensity_fit.py [hunk 1]
```diff
@@ -20,52 +20,43 @@ def test_chord_length():
...
 def test_muon_efficiency_fit(prod5_lst, reference_location):
-    from ctapipe.coordinates import CameraFrame, TelescopeFrame
+    from ctapipe.coordinates import TelescopeFrame
     from ctapipe.image.muon.intensity_fitter import (
         MuonIntensityFitter,
         image_prediction,
     )
     from ctapipe.instrument import SubarrayDescription
 
+    radius = 1.1959596 * u.deg
```
Comment (RuneDominik) [inline ctapipe/image/muon/tests/test_intensity_fit.py:30]: Overwritten before being used in l42 (with a slightly different value, 1.2 vs 1.1959596)

**Review by RuneDominik — CHANGES_REQUESTED** [review]
Given that this fixes #2467 and with that the effective vs. equivalent focal-length discrepancy with, according to the issue, up to 25% differences in the muon efficiencies, I'd argue that this needs a changelog entry.

**Comment by maxnoe** [comment]
@RuneDominik Yes, good point! I oversaw this change before.

**Review by RuneDominik — APPROVED** [review]

File: ctapipe/image/muon/intensity_fitter.py [hunk 2]
```diff
@@ -309,7 +308,8 @@ def image_prediction_no_units(
 
 def build_negative_log_likelihood(
     image,
-    telescope_description,
+    optics,
+    geometry_tel_frame,
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply ctapipe/image/muon/intensity_fitter.py:312]: The test is through the `__call__` method of the class, there is no direct test of this method.

**Review by Tobychev — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**